src/applications/disability-benefits/all-claims/tests/components/ComboBox.unit.spec.jsx

```diff
@@ -36,111 +40,375 @@ describe('ComboBox', () => {
+
+      fireEvent.click(document);
+
+      expect(input).to.have.value(searchTerm);
       expect(listbox).to.have.length(0);
     });
   });
+
+  describe('Keyboard Interactions', () => {
+    it('should highlight listbox items down the list on ArrowDown', () => {
+      const searchTerm = 'g';
+      const { getAllByRole, getByTestId } = render(<ComboBox {...props} />);
+
+      const input = getByTestId('combobox-input');
+      simulateInputChange(input, searchTerm);
+      const listboxItems = getAllByRole('option');
+      fireEvent.keyDown(input, { key: 'ArrowDown' });
+
+      expect(listboxItems[0]).to.have.class('cc-combobox__option--active');
+      expect(listboxItems[0]).to.have.attribute('aria-selected', 'true');
+
+      fireEvent.keyDown(input, { key: 'ArrowDown' });
+
+      expect(listboxItems[1]).to.have.class('cc-combobox__option--active');
+      expect(listboxItems[1]).to.have.attribute('aria-selected', 'true');
+    });
+
+    it('should stop at the last item in the listbox on repeated ArrowDown', () => {
+      const searchTerm = 'h';
+      const { getAllByRole, getByTestId } = render(<ComboBox {...props} />);
+
+      const input = getByTestId('combobox-input');
+      simulateInputChange(input, searchTerm);
+      const listboxItems = getAllByRole('option');
+
+      for (let i = 0; i < 24; i++) {
+        fireEvent.keyDown(input, { key: 'ArrowDown' });
+      }
+
+      expect(listboxItems[20]).to.have.class('cc-combobox__option--active');
+      expect(listboxItems[20]).to.have.attribute('aria-selected', 'true');
+    });
+
+    it('should highlight listbox items up the list on ArrowUp', () => {
+      const searchTerm = 'i';
+      const { getAllByRole, getByTestId } = render(<ComboBox {...props} />);
+
+      const input = getByTestId('combobox-input');
+      simulateInputChange(input, searchTerm);
+      const listboxItems = getAllByRole('option');
+      fireEvent.keyDown(input, { key: 'ArrowDown' });
+      fireEvent.keyDown(input, { key: 'ArrowDown' });
+
+      expect(listboxItems[1]).to.have.class('cc-combobox__option--active');
+      expect(listboxItems[1]).to.have.attribute('aria-selected', 'true');
+
+      fireEvent.keyDown(input, { key: 'ArrowUp' });
+
+      expect(listboxItems[0]).to.have.class('cc-combobox__option--active');
+      expect(listboxItems[0]).to.have.attribute('aria-selected', 'true');
+    });
+
+    // TODO: This test is working in the browser but here it's getting stuck on the free-text item
+    // it('should move focus to input on ArrowUp when at the top of the list', () => {
+    //   const searchTerm = 'i';
+    //   const { getByTestId } = render(<ComboBox {...props} />);
+
+    //   const input = getByTestId('combobox-input');
+    //   simulateInputChange(input, searchTerm);
+    //   fireEvent.keyDown(input, { key: 'ArrowDown' });
+    //   fireEvent.keyDown(input, { key: 'ArrowUp' });
+
+    //   expect(document.activeElement).to.eq(input);
+    // });
```

Comment:
There might be something off with your assertion `expect(document.activeElement).to.eq(input);`. I tried adding this line to the test below `should close the list and retain input on Escape` and the focus is going to the `body` when I would expect it would be the input...
<img width="1078" alt="Screenshot 2024-09-23 at 3 45 10 PM" src="https://github.com/user-attachments/assets/b591c059-3749-48d7-a55d-76d510f1c4af">